packages/core/upload/server/services/upload.js

```diff
@@ -86,7 +86,7 @@ module.exports = ({ strapi }) => ({
       caption: fileInfo.caption,
       folder: fileInfo.folder,
       folderPath: await fileService.getFolderPath(fileInfo.folder),
-      hash: generateFileName(basename),
+      hash: (await getService('provider').generateFileName(basename)) || generateFileName(basename),
```

Comment:
following the 1st comment then this can be simplified

```suggestion
      hash: await getService('provider').generateFileName(basename),
```